File: app/src/main/java/com/returnnotfound/stain/data/mapper/AnimeTopMapper.java
package com.returnnotfound.stain.data.mapper;

import com.returnnotfound.stain.data.dto.AnimeTopDTO;
import com.returnnotfound.stain.data.model.AnimeTop;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface AnimeTopMapper {
  AnimeTopMapper INSTANCE = Mappers.getMapper(AnimeTopMapper.class);

  AnimeTop toModel(AnimeTopDTO dto);
}

File: app/src/main/java/com/returnnotfound/stain/data/mapper/AnimeTopMapper.kt
package com.returnnotfound.stain.data.mapper

import com.returnnotfound.stain.data.dto.AnimeTopDTO
import com.returnnotfound.stain.data.model.AnimeTop
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper
interface AnimeTopMapper {
  fun toModel(dto: AnimeTopDTO?): AnimeTop?

  companion object {
    val INSTANCE = Mappers.getMapper(AnimeTopMapper::class.java)
  }
}